src/targets/gpu/device/layernorm.cpp

```diff
@@ -86,10 +94,13 @@ __device__ void layernorm(index_int i,
     const bool in_range    = idx.local < relements_v;
 
     auto mean = [&](auto z) {
-        return auto_block_reduce<MaxBlockSize>(
-                   idx, sum{}, value_type(0), relements_v, [=](auto) { return z; }) /
-               value_type(relements);
-
+        auto m = auto_block_reduce<MaxBlockSize>(
+                     idx, sum{}, value_type(0), relements_v, [=](auto) { return z; }) /
+                 value_type(relements);
+#if MIGRAPHX_WORKAROUND_NAVI_DPP_SYNC
+        __syncthreads();
```

Comment:
Some other things to try for a workaround is `__builtin_amdgcn_s_barrier`, `__threadfence`, or `__threadfence_block`.

Reply:
looks like `__builtin_amdgcn_s_barrier` passes correctness while improving workaround performance.